modules/button/react/lib/IconButton.tsx

```diff
@@ -116,6 +113,25 @@ IconButton.Size = {
 
 export default IconButton;
 
+const getIconButtonBorderRadius = (variant: IconButtonVariant) => {
+  switch (variant) {
+    case IconButtonVariant.Square:
+    case IconButtonVariant.SquareFilled:
+    case IconButtonVariant.ToolbarSquare:
+      return {borderRadius: borderRadius.m};
+    default:
+      return {borderRadius: borderRadius.circle};
+  }
+};
+
+const getIconButtonSize = (variant: IconButtonVariant, size: 'small' | 'medium') => {
+  if (variant === IconButtonVariant.ToolbarSquare) {
```

Comment:
I'm hesitant to do this. A big part of the v4 button refactor was to avoid API restrictions like this where some combinations aren't valid (i.e. a medium toolbar icon button).

I think we should either allow medium toolbar icon buttons, or this should be broken into it's own `ToolbarIconButton` component

Reply:
Yeah, agreed. In the example I suggested:
```js
  if (size) {
    // always allow an override
    return size;
  }
```

That allows us some flexibility to potentially add a medium sized toolbar icon without worrying about breaking changes down the road.

The other option is to create a separate component here, which is fine too.